src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.Array.cs

```diff
@@ -33,13 +32,19 @@ public void WriteEndArray()
                 EnsureWriteCapacity(1);
                 _buffer[_offset++] = CborInitialByte.IndefiniteLengthBreakByte;
             }
+
+            AdvanceDataItemCounters();
         }
 
         public void WriteStartArrayIndefiniteLength()
         {
+            if (CborConformanceLevelHelpers.RequiresDefiniteLengthItems(ConformanceLevel))
+            {
+                throw new InvalidOperationException("Indefinite-length items are not permitted under the current conformance level.");
```

Comment:
@bartonjs would using NotSupportedException be more appropriate in this context?

Reply:
`NotSupportedException` is generally used as the entire body of a method that doesn't make sense for a class that it's required to have because of a base class, interface, or compiler-recognized pattern.  Less often it's used for "you're in the wrong mode", e.g. a FileStream will report false for CanWrite if it was opened read-only, and Stream.Write will throw an NSE if CanWrite is false.

But for InvalidOperationException:

* DO throw an InvalidOperationException if the object is in an inappropriate state.

The state of this object is to write per some conformance mode, and that conformance mode says no indefinite arrays... so InvalidOperationException seems correct to me.